config/src/config/logger_config.rs

```diff
@@ -51,3 +56,66 @@ impl LoggerConfig {
         self.tokio_console_port = Some(utils::get_available_port());
     }
 }
+
+impl ConfigSanitizer for LoggerConfig {
+    /// Validate and process the logger config according to the given node role and chain ID
+    fn sanitize(
+        node_config: &mut NodeConfig,
+        _node_role: RoleType,
+        _chain_id: ChainId,
+    ) -> Result<(), Error> {
+        let sanitizer_name = Self::get_sanitizer_name();
+        let logger_config = &node_config.logger;
+
+        // Verify that tokio console tracing is correctly configured
+        if is_tokio_console_enabled() && logger_config.tokio_console_port.is_none() {
+            return Err(Error::ConfigSanitizerFailed(
+                sanitizer_name,
+                "The tokio-console feature is enabled but the tokio console port is not set!"
+                    .into(),
+            ));
+        } else if !is_tokio_console_enabled() && logger_config.tokio_console_port.is_some() {
+            return Err(Error::ConfigSanitizerFailed(
+                sanitizer_name,
+                "The tokio-console feature is not enabled but the tokio console port is set!"
```

Comment:
another maybe-too picky spot? It's not bad to have it disabled but configured in case of turning it back on temporarily for diagnostics?

Reply:
Potentially. @ibalajiarun actually mentioned something similar too (e.g., see here for my reasoning: https://github.com/aptos-labs/aptos-core/pull/7781). I think the best approach is to probably change the default based on the `tokio-console` feature (i.e., it's it's enabled we set the config to `Some(...)`, otherwise, we set it to `None`). And this kind of fine-grained tuning is coming soon (my next PR?), so hopefully we can get the best of both worlds once this lands.